code/src/cljs/sixsq/nuvla/ui/main/effects.cljs

```diff
@@ -24,3 +24,22 @@
         (stripejs/loadStripe)
         (.then (fn [stripe]
                  (callback stripe))))))
+
+
+(defn- before-unload-handler [event]
+  (set! (.-returnValue event) "")
+  (.preventDefault event))
+
+(defn- clear-unload-protection []
+  (.removeEventListener js/window "beforeunload" before-unload-handler))
```

Comment:
I don't understand this one, because I do not call `removeEventListener` with a third argument.

The docu says the third parameter is optional when calling `window.removeEventListener` or `window.addEventListener`, but `removeEventListener` has to be called with the same arguments as `addEventListener`, else it fails to remove the event listener.

Do you mean the naming `-handler` instead of `-listener`?

Reply:
You are right, I misunderstand the documentation. You can consider this one as done